toptal/davinci-github-actions
# Clean up tmp-workspaces file in the yarn install action

### Description

The yarn install action creates a temporary tmp-workspaces.json file which stays in the root of the repository after the action is run and pollutes the workspace.

It breaks the actions that create PRs out of the generated changes like changesets action. This file is [ignored in davinci](https://github.com/toptal/davinci/blob/d6b3bd60b5170f31eb6b8cd1c0ee2f097b2665f5/.gitignore#L24), will be ignored in Talent Portal and all repos that implement publishing with changesets.

The file is no needed after it contributes to the cache hash.

In case of single repository (not monorepo) the file didn't exist as there was a typo which means the workflow works even without this file.

### How to test

- Run the modified action in workflows of monorepo and see no tmp-workspaces.json file left in the root of the repo.

### Development checks


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>
